File: Java_Assignment/Assignment_46/programAQ4.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement46Question4.java
//  Description : Write Java program which accept number of rows and number of columns
//                from user and display below pattern.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*
Input : iRow = 4 iCol = 5
Output : 4 4 4 4 4
3 3 3 3 3
2 2 2 2 2
1 1 1 1 1

Program Layout :
*/

import java.util.Scanner;

class Pattern
{
    public void PatternX(int iRow,int iCol)
    {   
        int i = 0, j = 0;

        for(i = iRow;i >= 1;i--)
        {
            for(j = 1;j <= iCol;j++)
            {
                System.out.print(i + "\t");
            }
            System.out.println();
        }
    }
}

public class programAQ4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        int iValue1 = 0, iValue2 = 0;

        System.out.println("Enter number of Row: ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter number of Columns: ");
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.PatternX(iValue1, iValue2);

        sobj.close();
    }   
}

File: C_Programming Assignment/Assignment_14/programAQ4.c
///////////////////////////////////////////////////////////
//
//  File name : Assigment14Question4.c
//  Descreption : Accept number of rows and number of columns from user and display below pattern
//  Author : Kartik Ganesh Jare
//  Date : 26/5/25
//  
///////////////////////////////////////////////////////////

/* Input : iRow = 5 iCol = 5
Output : 1 2 3 4 5
        -1 -2 -3 -4 -5 
         1 2 3 4 5
        -1 -2 -3 -4 -5   
*/

#include<stdio.h>

void Pattern(int iRow,int iCol)
{   
    int i = 0,j = 0,iNo1 = 0,iNo2 = 0;

    for(i = 1 ; i <= iRow; i++)
    {
        for(j = 1,iNo1 = 1,iNo2 = -1; j <= iCol; j++,iNo1 ++,iNo2--)
        {
            if(i % 2 == 0)
            {
                printf("%d\t",iNo2);
            }
            else
            {
                printf("%d\t",iNo1);
            }
        }
        printf("\n");
    }
        
} 

int main()
{
    int iValue1 = 0, iValue2 = 0; 
    
    printf("Enter number of rows and columns : \n");
    scanf("%d %d",&iValue1,&iValue2);

    Pattern(iValue1,iValue2);

    return 0;
}
    